flink-table/flink-table-planner/pom.xml

```diff
@@ -308,6 +309,20 @@ under the License.
 				</executions>
 			</plugin>
 
+			<plugin>
+				<groupId>org.apache.maven.plugins</groupId>
+				<artifactId>maven-compiler-plugin</artifactId>
+				<configuration>
+					<annotationProcessorPaths>
+						<path>
+							<groupId>org.immutables</groupId>
+							<artifactId>value</artifactId>
+							<version>${immutables.version}</version>
+						</path>
+					</annotationProcessorPaths>
+				</configuration>
+			</plugin>
```

Comment:
the reason was mentioned at https://inside.java/2024/06/18/quality-heads-up/

so now either use proc:full all explicitly put path to annotation processors
same approach as for lombok